packages/core/src/modules/discover-features/protocol/v2/V2DiscoverFeaturesService.ts

```diff
@@ -0,0 +1,43 @@
+import type { FeatureQueryOptions } from '../../models'
+
+import { Dispatcher } from '../../../../agent/Dispatcher'
+import { injectable } from '../../../../plugins'
+import { FeatureRegistry } from '../../FeatureRegistry'
+
+import { V2DisclosuresMessageHandler, V2QueriesMessageHandler } from './handlers'
+import { V2QueriesMessage, V2DisclosuresMessage } from './messages'
+
+@injectable()
+export class V2DiscoverFeaturesService {
+  private featureRegistry: FeatureRegistry
+
+  public constructor(dispatcher: Dispatcher, featureRegistry: FeatureRegistry) {
+    this.featureRegistry = featureRegistry
+
+    this.registerHandlers(dispatcher)
+  }
+
+  private registerHandlers(dispatcher: Dispatcher) {
+    dispatcher.registerHandler(new V2DisclosuresMessageHandler())
+    dispatcher.registerHandler(new V2QueriesMessageHandler(this))
+  }
+
+  public async createQuery(options: { queries: FeatureQueryOptions[] }) {
```

Comment:
Ah I see it should be plural here. Maybe we should call this createQueries or something? I am not too sure which one it has to be, but can we make them consistent?